### PR TITLE
Add cookie settings page

### PR DESCRIPTION
**Pull request by vanitabarrett**
New page at `/help/cookies` which will allow users to adjust their cookie preferences (opt in/out of the different categories)
 
Screenshot_2019-06-04 Cookie settings on GOV UK

The page needs to be published as a special route using the `publish_special_routes` rake task - this will be done after this PR is merged. 

Page can be viewed here: https://new-cookie-banner.herokuapp.com/help/cookies

### CODE REVIEW

**Comment by vanitabarrett** [comment]
[Productionize the new cookie settings page](https://trello.com/c/fADkc6L5/160-productionize-the-new-cookie-settings-page)

File: spec/javascripts/unit/modules/cookie-settings.spec.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        } else {
+          expect(radioButtons[i].checked).not.toBeTruthy()
+        }
+      }
+    });
+  });
+
+  describe('submitSettingsForm', function() {
+    beforeEach(function() {
+      confirmationContainer = document.createElement('div')
```
Comment (chao-xian) [inline spec/javascripts/unit/modules/cookie-settings.spec.js]: We should be able to DRY these up from here and `showConfirmationMessage` into the parent `beforeEach`

File: app/assets/stylesheets/views/_cookie-settings.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.cookie-settings__confirmation {
+  @include core-19;
```
Comment (alex-ju) [inline app/assets/stylesheets/views/_cookie-settings.scss]: Is there a reason for not using the new mixins? `@include govuk-font(19);`

File: app/views/help/cookie_settings.html.erb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          text: t('cookies.off')
+        }
+      ]
+    } %>
+
+    <%= render "govuk_publishing_components/components/radio", {
+      name: "cookies-campaigns",
+      inline: true,
+      heading: t('cookies.types.campaigns'),
+      hint: "These cookies are set by third party websites and do things like:
```
Comment (alex-ju) [inline app/views/help/cookie_settings.html.erb]: We could avoid `html_safe` here by capturing this block above, which could improve the markup readability too:

```
<% cookies_campaigns_hint = capture do %>
  <ul>...</ul>
<% end %>
```

File: app/assets/javascripts/modules/cookie-settings.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  CookieSettings.prototype.showConfirmationMessage = function () {
+    var confirmationMessage = document.querySelector('div[data-cookie-confirmation]')
+    var previousPageLink = document.querySelector('.cookie-settings__prev-page')
+    var referrer = CookieSettings.prototype.getReferrerLink()
+
+    document.body.scrollTop = document.documentElement.scrollTop = 0
+
+    if (referrer && referrer !== '/help/cookies') {
```
Comment (alex-ju) [inline app/assets/javascripts/modules/cookie-settings.js]: Is there a way to read this path from markup (a data attribute for example) so we don't need to keep it updated in two places?

File: app/assets/javascripts/modules/cookie-settings.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    return false
+  }
+
+  CookieSettings.prototype.showConfirmationMessage = function () {
+    var confirmationMessage = document.querySelector('div[data-cookie-confirmation]')
+    var previousPageLink = document.querySelector('.cookie-settings__prev-page')
+    var referrer = CookieSettings.prototype.getReferrerLink()
+
+    document.body.scrollTop = document.documentElement.scrollTop = 0
```
[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/assets/javascripts/modules/cookie-settings.js]: @alex-ju The confirmation message does have a wrapper around it with `aria-live=assertive`, so it should read out the message when the form is submitted (for those using screen readers).

https://github.com/alphagov/frontend/pull/1856/files#diff-e394cdb6571fb405da3283b4f5d5cfb8R4

File: app/assets/stylesheets/views/_cookie-settings.scss (hunk 2 above)
Comment (vanitabarrett) [reply app/assets/stylesheets/views/_cookie-settings.scss]: @alex-ju No, just failing to break out of old habits! Will correct 👍

File: app/assets/javascripts/modules/cookie-settings.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/assets/javascripts/modules/cookie-settings.js]: Ah, I've missed that one. It doesn't seem to do the job without moving the focus there (tested with VoiceOver on Safari and Chrome). Can we do some testing around that, if we haven't already.

[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/assets/javascripts/modules/cookie-settings.js]: @alex-ju Hmm strange, I think it worked when I tested it before. I'll take a look 🙂

File: app/assets/javascripts/modules/cookie-settings.js (hunk 4 above)
Comment (vanitabarrett) [reply app/assets/javascripts/modules/cookie-settings.js]: @alex-ju I've changed the logic slightly instead, so it now checks that the referrer isn't equal to `document.location.pathname` (e.g: the current URL).

Comment (alex-ju) [reply app/assets/javascripts/modules/cookie-settings.js]: cool, even better

File: app/assets/javascripts/modules/cookie-settings.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/assets/javascripts/modules/cookie-settings.js]: @alex-ju I've got a PR here for adding the ability to add an `aria-live` attribute to the notice component https://github.com/alphagov/govuk_publishing_components/pull/911

**Comment by vanitabarrett** [comment]
I've now added the `aria_live: true` flag to the notice component. https://github.com/alphagov/frontend/pull/1856/files#diff-e394cdb6571fb405da3283b4f5d5cfb8R7

This won't come into effect until `govuk_publishing_components` has been versioned and bumped in this PR, but I'm holding off doing that for now until we're ready to go live as we have more changes in the gem which we need before going live anyway. cc @alex-ju

**Comment by vanitabarrett** [comment]
[GO LIVE](https://trello.com/c/TG1lBZbk/168-go-live)

**Pull request merged**